duplicate.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
```

Comment:
Let's call the file "similarity.py" instead.

We'll have to refactor it a bit to make it more generic and make it easy to test other algorithms, but we can do it later.